hw/xbox/xbox.c

```diff
@@ -253,7 +253,53 @@ void nv2a_init(PCIBus *bus, int devfn, MemoryRegion *ram);
 /* PC hardware initialisation */
 static void xbox_init(MachineState *machine)
 {
-    xbox_init_common(machine, default_eeprom, NULL, NULL);
+    char *filename;
+    int fd;
+    int rc;
+    int eeprom_file_size;
+    const int eeprom_size = 256;
+    uint8_t *eeprom_data = g_malloc(eeprom_size);
+
+    const char *eeprom_file = object_property_get_str(qdev_get_machine(),
+                                                      "eeprom", NULL);
+    if ((eeprom_file != NULL) && *eeprom_file) {
```

Comment:
nit: is `*eeprom_file` really necessary and the QEMU way of doing things?

Reply:
Yes, it is necessary. What are you suggesting?

Reply:
`strlen(eeprom_file) > 0` or something?

I think it's just not very self-documenting.

Reply:
Ah yes, agree